scripts/ia_healthcheck.py

```diff
@@ -0,0 +1,143 @@
+import requests
+from requests.auth import HTTPBasicAuth
+from datetime import datetime, timedelta
+import os
+import random
+import sys
+
+MAX_CAPTURE_AGE = timedelta(hours=72)
+LINKS_TO_CHECK = 5
+SCANNER_USER = os.environ['SCANNER_USER']
+SCANNER_PASSWORD = os.environ['SCANNER_PASSWORD']
+
+def query_webmondb():
+    """
+    Set user and pass as environmental variables from command line
+    For MacOS: in the terminal:
+    export SCANNER_USER="your API user name"
+    export SCANNER_PASSWORD="your API password"
+    Query the web monitoring API to get random links
+    Integer -> List
+    """
+    api = 'https://api.monitoring.envirodatagov.org/api/v0/pages?chunk_size=1'
+    try:
+        response = requests.get(api, auth=HTTPBasicAuth(SCANNER_USER, SCANNER_PASSWORD))
+    except Exception as e:
+        print('Could not access Scanner API')
+        print(e)
+        return []
+    else:    
+        response = response.json()
+        url_count = response['meta']['total_results']
+        links = [query_webmondb_url(number) for number in random.sample(range(url_count),LINKS_TO_CHECK)]
+        return links
+
+def query_webmondb_url(number):
+    """
+    Take an integer and query the web-mon-api to retreive a url based on n urls.
+    The integer is the chunk number of chunk_size of 1.   
+    Int -> Str
+    """
+    api = 'https://api.monitoring.envirodatagov.org/api/v0/pages?chunk='+str(number)+'&chunk_size=1'
+    try:
+        response = requests.get(api, auth=HTTPBasicAuth(SCANNER_USER, SCANNER_PASSWORD))
+    except Exception as e:
+        print('Could not access Scanner API with link: %d' % number)
+        print(e)
+        return None
+    else:
+        response = response.json()
+        return response['data'][0]['url']     
+
+def query_wayback(url):
```

Comment:
Our python tools also have a [nice module for querying wayback](https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/master/web_monitoring/internetarchive.py). It uses Python `datetime` objects and does all the parsing and formatting of responses, too (even if you might not need that part):

```py
from web_monitoring.internetarchive import list_versions

def wayback_has_snapshot(url, after_date):
    try:
        return len(list_versions(url, from_date=after_date)) > 0
    except ValueError:
        # list_versions raises a ValueError when there are no records
        return False
```

There’s also a lower-level querying API that doesn’t do as much fancy stuff with your input, still does all the output parsing:

```py
from web_monitoring.internetarchive import search_cdx

def wayback_has_snapshot(url, after_date):
    snapshots = search_cdx({
        'url': url,
        'from': after_date.strftime('%Y%m%d%H%M%S'),
        # If you wanted to only make sure Wayback captured successful
        # responses, that’d be a good reason to use this lower-level API:
        # 'filter': 'statuscode:200'
    })
    return len(list(snapshots)) > 0
```

The response objects from either of those two methods have lots of useful properties if you wanted to add extra metadata to the output:

- `key` (Sortable URL Key)
- `timestamp` (Wayback timestamp string, see `date` below for parsed version)
- `url` (Actual URL of the page; should roughly equal the URL you requested)
- `mime_type` (If available, the MIME type of the URL, e.g. `text/html`)
- `status_code` (HTTP status code Wayback got when it 
- `digest` (Base 32 encoded SHA-1 of the raw content)
- `length` (Length, in bytes, of the raw content)
- `date` (Parsed timestamp as a python `datetime`)
- `raw_url` (URL of raw response content in Wayback machine)
- `view_url` (URL of viewable-in-browser content in Wayback machine)

Reply:
Published docs: https://edgi-govdata-archiving.github.io/web-monitoring-processing/cdx_api.html